### PR TITLE
fix: close assigned issue logic

### LINKED ISSUE #23: Close Assigned Issues
We do not want assigned issues to render on work.ubq.fi 

The simple solution would be to close them (that way they would not be categorized as an "open" task)

The more complex solution would be to have special logic in the UI for the `unavailable` label but I feel that closing as not planned might be better? Then we can close as complete if it is actually closed as complete? 

> > However, the original issue was resolved according to the spec here and that was just to remove the `unavailable` bounties from the devpool repo was it not? 
> 
> You're right. I read the following as "closing" the issue which means that it would not render in the work.ubq.fi UI
> 
> > if bounty is unavailable (i.e. assigned to some bounty hunter) and exists in the list with the state open then close such bounty
> 
> With an emphasis on 
> 
> > then close such bounty
> 
> So we need to file a new issue and get a new pull request to actually close them?

_Originally posted by @pavlovcik in https://github.com/ubiquity/devpool-directory-bounties/issues/11#issuecomment-2017567237_

### PR DESCRIPTION
**Pull request by Ekep-Obasi**
### Solution

I realized that the existing logic for closing assigned issues works correctly, but the logic for updating issue always reopens the issue again

```js 
        if (
          devpoolIssue.title !== projectIssue.title ||
          devpoolIssue.state !== projectIssue.state || // <-- since the states are different, if block runs
          (!isDevpoolUnavailableLabel && projectIssue.assignee?.login) ||
          (isDevpoolUnavailableLabel && !projectIssue.assignee?.login) ||
          devpoolIssue.body !== projectIssue.html_url ||
          devpoolIssueLabelsStringified !== projectIssueLabelsStringified
        ) {
          await octokit.rest.issues.update({
            owner: DEVPOOL_OWNER_NAME,
            repo: DEVPOOL_REPO_NAME,
            issue_number: devpoolIssue.number,
            title: projectIssue.title,
            body,
            state: projectIssue.state as "open" | "closed", // <--- projectIssue state is set back to "open"
            labels: getDevpoolIssueLabels(projectIssue, projectUrl),
          });
          console.log(`Updated: ${devpoolIssue.html_url} (${projectIssue.html_url})`);
        } else {
          console.log(`No updates: ${devpoolIssue.html_url} (${projectIssue.html_url})`);
        }
``` 

You can see this behaviour on this [issue](https://github.com/ubiquity/devpool-directory/issues/1154) or on the screenshot below

evidence

### Change 

- [x] refactored the logic that updates issue 

```js 
// reopen issue only if it doesn't have the `Unavaliable` label 
state: !isDevpoolUnavailableLabel ? (projectIssue.state as "open" | "closed") : "closed",
``` 

- [x] added tests to assert that current functionality (closing of assigned issues) works properly and test are passing locally

work ubi test

### CODE REVIEW

**Review by rndquu — CHANGES_REQUESTED** [review]
The issue persists

Current behaviour when project issue is assigned and devpool issue is open:
- 1st `index.ts` run closes devpool issue
- 2nd `index.ts` run opens devpool issue
- 3rd (and all other) `index.ts` run closes devpool issue

**Comment by gentlementlegen** [comment]
After locally testing, the test `test` fails with the following state
```JSON
Error: expect(received).toMatchObject(expected)

- Expected  - 3
+ Received  + 9

@@ -22,11 +22,11 @@
        "subscriptions_url": "",
        "type": "",
        "url": "",
      },
      "author_association": "NONE",
-     "body": "body",
+     "body": "https://github.com/ubiquity/test-repo/issues/1",
      "closed_at": null,
      "comments": 0,
      "comments_url": "",
      "created_at": "",
      "events_url": "",
@@ -35,23 +35,29 @@
      "labels": Array [
        Object {
          "name": "Pricing: 200 USD",
        },
        Object {
-         "name": "Time: 1h",
+         "name": "Partner: ubiquity/test-repo",
        },
        Object {
          "name": "id: 1",
+       },
+       Object {
+         "name": "Unavailable",
        },
+       Object {
+         "name": "Time: 1h",
+       },
      ],
      "labels_url": "",
      "locked": false,
      "milestone": null,
      "node_id": "1",
      "number": 1,
      "repository_url": "https://github.com/ubiquity/devpool-directory",
-     "state": "closed",
+     "state": "open",
      "title": "issue",
      "updated_at": "",
      "url": "",
      "user": null,
    },
```

File: tests/main.test.ts [hunk 1]
```diff
@@ -106,4 +106,30 @@ describe("Devpool Updates", () => {
...
+    });
+    db.issue.create({
+      ...issueTemplate,
+      id: 2,
+      assignee: { ...issueTemplate.assignee, login: "john doe" },
+      owner: DEVPOOL_OWNER_NAME,
+      repo: "test-repo",
+    });
+    await main();
+    const devpoolIssue = await getAllIssues(DEVPOOL_OWNER_NAME, DEVPOOL_REPO_NAME);
```
Comment (gentlementlegen) [inline tests/main.test.ts:125]:
```suggestion
    const devpoolIssues = await getAllIssues(DEVPOOL_OWNER_NAME, DEVPOOL_REPO_NAME);
```

**Review by gentlementlegen — CHANGES_REQUESTED** [review]

**Comment by Ekep-Obasi** [comment]
I'm closing this PR, I think this issue had already been solved by @rndquu

**Pull request closed without merging**